Fixes and updates to make the module work

razvand opened the pull request:
Add configuration fixes to make the `attack_surface_approximation` module work. Improve instructions on using the module.

razvand committed:
configuration: Remove unused class GhidraDecompilation

The `GhidraDecompilation` class is not used anywhere. Remove it from the
`configuration.py` file.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
dict_gen: Handle case of error opening man file

There may be cases when there is a missing symlink or a man page file
is not accessible. Handle exception of opening that file: do nothing,
return from the generator function.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
Update pyprojects.toml with proper package versions

Add reference to the `pyprojects.toml` file in the `commons` repository.

The `pyelftools` v0.28 package version results in build error (unable to
locate `RelrRelocationSection`). It's not part of the `relocation.py`
file: https://github.com/eliben/pyelftools/blob/v0.28/elftools/elf/relocation.py

Version v0.29 of `pyelftools` defines `RelrRelocationSection`: https://github.com/eliben/pyelftools/blob/v0.29/elftools/elf/relocation.py#L110

Update the version of `pyelftools` in `pyprojects.toml`. Meaning using
the line below in `pyprojects.toml`:

```
pyelftools = "^0.29"
```

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
Add poetry.lock to repository

It is generally recommended to include `poetry.lock` to the repository.
This helps with using the same package versions that are known to work.

The absence of the `poetry.lock` file will force Poetry to generate a
new one, that may result in package versions that fail to wor

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
README: Improve instructions

Improve instructions in `README.md` on installing and setting up the
module. Add detailed instructions, fix commands where required,
reference the `dataset` and `commons` modules.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

ClaraStefania commented on README.md:
```diff
@@ -23,122 +23,179 @@
 
-The module works by automating Ghidra for static binary analysis. It extracts information and apply heuristics to determine if a given input stream is present.
+The module works by automating [Ghidra](https://ghidra-sre.org/) for static binary analysis.
+It extracts information and apply heuristics to determine if a given input stream is present.
```
```suggestion
It extracts information and applies heuristics to determine if a given input stream is present.
```

ClaraStefania commented on README.md:
````diff
@@ -23,122 +23,179 @@
+     which poetry
+     ```
+
+     you should get something a path ending with `.venv/bin/poetry`.
````
```suggestion
     you should get a path ending with `.venv/bin/poetry`.
```

ClaraStefania commented on attack_surface_approximation/dictionaries_generators/heuristics/man_parsing.py:
The last line, 'return all_arguments', should be deleted. It is written twice and causes the error
```
SyntaxError: 'return' outside function
```

ClaraStefania reviewed (commented):
I suggested a few small changes, one that solves an error and two in the README file.

ClaraStefania commented on README.md:
````diff
@@ -23,122 +23,179 @@
+#### Fuzz Arguments
+
+```console
+$ poetry run attack_surface_approximation fuzz  $(pwd)/../dataset/executables/toy_test_suite_1.elf --dictionary args.txt
````
```suggestion
$ poetry run attack_surface_approximation fuzz --elf $(pwd)/../dataset/executables/toy_test_suite_1.elf --dictionary args.txt
```

razvand committed:
poetry: Upgrade vulnerable version of package "black"

Use version `^24.3.0` to fix vulnerability in the dependabot alert #1
(on GitHub).

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
man_parsing: Remove redundant return statement

A `return` line is present by mistake outside the scope of a function.
Remove it.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

AmaliaI12 commented on README.md:
````diff
@@ -23,122 +23,179 @@
+   export PYTHON_KEYRING_BACKEND=keyring.backends.null.Keyring
+   ```
+
+   This is an problem that may occur in certain situations, preventing Poetry from getting packages.
````
```suggestion
   This is a problem that may occur in certain situations, preventing Poetry from getting packages.
```
